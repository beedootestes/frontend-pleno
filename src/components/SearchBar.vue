<template>
  <div class="search-bar">
    <img src="../assets/beebot.png" alt="Beebot logo">
    <input type="text" placeholder="Pergunte para o Beebot" />
    <div class="mic-container">
      <MicIcon :size="32" />
    </div>
  </div>
</template>
<script setup lang="ts">
import MicIcon from 'vue-material-design-icons/Microphone.vue';

</script>
<style scoped lang="scss">
  .search-bar {
    width: 100%;
    border: 2px solid #dadada;
    border-radius: 31px;
    display: flex;
    align-items: center;
    padding: 5px 13px 5px 20px;
    box-sizing: border-box;

    input {
      background-color: white;
      outline: none;
      color: #000;
      border: none;
      width: 100%;
      height: 100%;
      box-sizing: border-box;
    }

    .mic-container {
      background-color: $color-active;
      border-radius: 50%;
      box-sizing: border-box;
      color: white;
      padding: 7px;
    }
  }
</style>