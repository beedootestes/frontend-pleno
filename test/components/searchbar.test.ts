import {describe, expect, test} from "vitest";
import SearchBar from "../../src/components/SearchBar.vue";
import {mount, VueWrapper} from "@vue/test-utils";


describe('Test SearchBar', () => {
  test('mount component', async () => {
    expect(SearchBar).toBeTruthy()

    const wrapper: VueWrapper<SearchBar> = mount(SearchBar)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
