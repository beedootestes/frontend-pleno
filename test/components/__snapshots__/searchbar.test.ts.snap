// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test SearchBar > mount component 1`] = `
"<div data-v-6ab8b910=\\"\\" class=\\"search-bar\\"><img data-v-6ab8b910=\\"\\" src=\\"/src/assets/beebot.png\\" alt=\\"Beebot logo\\"><input data-v-6ab8b910=\\"\\" type=\\"text\\" placeholder=\\"Pergunte para o Beebot\\">
  <div data-v-6ab8b910=\\"\\" class=\\"mic-container\\"><span data-v-6ab8b910=\\"\\" aria-hidden=\\"true\\" class=\\"material-design-icon microphone-icon\\" role=\\"img\\"><svg fill=\\"currentColor\\" class=\\"material-design-icon__svg\\" width=\\"32\\" height=\\"32\\" viewBox=\\"0 0 24 24\\"><path d=\\"M12,2A3,3 0 0,1 15,5V11A3,3 0 0,1 12,14A3,3 0 0,1 9,11V5A3,3 0 0,1 12,2M19,11C19,14.53 16.39,17.44 13,17.93V21H11V17.93C7.61,17.44 5,14.53 5,11H7A5,5 0 0,0 12,16A5,5 0 0,0 17,11H19Z\\"><!--v-if--></path></svg></span></div>
</div>"
`;
